cpp/velox/tests/VeloxColumnarToColumnarTest.cc

```diff
@@ -0,0 +1,100 @@
+#include "operators/serializer/VeloxColumnarToColumnarConverter.h"
+#include "utils/VeloxArrowUtils.h"
+#include "velox/vector/arrow/Bridge.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+using namespace facebook;
+using namespace facebook::velox;
+
+namespace gluten {
+class VeloxColumnarToColumnarTest : public ::testing::Test, public test::VectorTestBase {
+ protected:
+  static void SetUpTestCase() {
+    memory::MemoryManager::testingSetInstance({});
+  }
+
+  void testRowVectorEqual(ArrowSchema cSchema, ArrowArray arrowArray, velox::RowVectorPtr vector) {
+    auto columnarToColumnarConverter = std::make_shared<VeloxColumnarToColumnarConverter>(&cSchema, pool_);
+
+    auto cb = columnarToColumnarConverter->convert(&arrowArray);
+    auto vp = std::dynamic_pointer_cast<VeloxColumnarBatch>(cb)->getRowVector();
+    velox::test::assertEqualVectors(vector, vp);
+  }
+
+ private:
+  std::shared_ptr<arrow::MemoryPool> arrowPool_ = defaultArrowMemoryPool();
+};
+
+TEST_F(VeloxColumnarToColumnarTest, allTypes) {
+  auto vector = makeRowVector({
+      makeNullableFlatVector<int8_t>({1, 2, 3, std::nullopt, 4, std::nullopt, 5, 6, std::nullopt, 7}),
+      makeNullableFlatVector<int8_t>({1, -1, std::nullopt, std::nullopt, -2, 2, std::nullopt, std::nullopt, 3, -3}),
+      makeNullableFlatVector<int32_t>({1, 2, 3, 4, std::nullopt, 5, 6, 7, 8, std::nullopt}),
+      makeNullableFlatVector<int64_t>(
+          {std::nullopt,
+           std::nullopt,
+           std::nullopt,
+           std::nullopt,
+           std::nullopt,
+           std::nullopt,
+           std::nullopt,
+           std::nullopt,
+           std::nullopt,
+           std::nullopt}),
+      makeNullableFlatVector<float>(
+          {-0.1234567,
+           std::nullopt,
+           0.1234567,
+           std::nullopt,
+           -0.142857,
+           std::nullopt,
+           0.142857,
+           0.285714,
+           0.428617,
+           std::nullopt}),
+      makeNullableFlatVector<bool>(
+          {std::nullopt, true, false, std::nullopt, true, true, false, true, std::nullopt, std::nullopt}),
+      makeFlatVector<velox::StringView>(
+          {"alice0", "bob1", "alice2", "bob3", "Alice4", "Bob5", "AlicE6", "boB7", "ALICE8", "BOB9"}),
+      makeNullableFlatVector<velox::StringView>(
+          {"alice", "bob", std::nullopt, std::nullopt, "Alice", "Bob", std::nullopt, "alicE", std::nullopt, "boB"}),
+  });
+
+  ArrowSchema cSchema;
+  ArrowArray arrowArray;
+  exportToArrow(vector, cSchema, ArrowUtils::getBridgeOptions());
+  exportToArrow(vector, arrowArray, pool(), ArrowUtils::getBridgeOptions());
+  testRowVectorEqual(cSchema, arrowArray, vector);
+}
+
+TEST_F(VeloxColumnarToColumnarTest, decimal) {
+  auto vector = makeRowVector({
+      makeNullableFlatVector<int64_t>({1.00, 2.00, 3.00}, DECIMAL(5, 2))
+  });
+
+  ArrowSchema cSchema;
+  ArrowArray arrowArray;
+  exportToArrow(vector, cSchema, ArrowUtils::getBridgeOptions());
```

Comment:
If I'm understanding well, there might be a bug that velox doesn't handle short decimal well when converting `arrowArray`-> `rowVector`

In `exportToArrow`, we can see it needs to convert to `int128_t` since it's arrow short decimal.

```c++
rows.apply([&](vector_size_t i) {
      int128_t value = buf.as<int64_t>()[i];
      memcpy(dst + (j++) * sizeof(int128_t), &value, sizeof(int128_t));
    });
```

But in `importFromArrow`, velox directly handle it as `int64_t`, whereas this test won't pass.

```c++
values = wrapInBufferView(
        arrowArray.buffers[1], arrowArray.length * type->cppSizeInBytes());
```

cc @PHILO-HE @zhouyuan pls correct me if I'm wrong

Reply:
+1 looks a bug

Reply:
I see, let me try to fix velox firstly

Reply:
fix it here: https://github.com/facebookincubator/velox/pull/8957